File: engine/src/main/java/edu/austral/dissis/chess/engine/CheckMate.java
package edu.austral.dissis.chess.engine;

public class CheckMate implements WinCondition{

  @Override
  public boolean winCondition(Board board, Color color) {
    //if (board.getPiece().isAlive() == false){
    //  System.out.println("CheckMate");
    return false;
    //}
  }
}

File: engine/src/main/kotlin/edu/austral/dissis/chess/engine/winCondition/CheckMate.kt
package edu.austral.dissis.chess.engine.winCondition

import edu.austral.dissis.twoDBoardGame.board.Board
import edu.austral.dissis.twoDBoardGame.rules.RuleManager
import edu.austral.dissis.twoDBoardGame.winCondition.WinCondition
import java.awt.Color

class CheckMate: WinCondition {
 // val checkValidator = ()

  override fun checkWinner(board: Board, color: Color, gameRules: List<RuleManager>): Boolean {
    TODO("Not yet implemented")
  }
}